### PR TITLE
add a connector sample

### PR DESCRIPTION
**Pull request by freeznet**
Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

- add connector image sample
- fix runner base build error
- specific platform when build

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: images/samples/build.sh [hunk 1]
```diff
@@ -21,26 +21,32 @@
...
 docker tag ${PYTHON_SAMPLE} "${DOCKER_REPO}"/${PYTHON_SAMPLE}:"${PULSAR_IMAGE_TAG}"
 
 echo "build go sample"
-docker build -t ${GO_SAMPLE} images/samples/go-function-samples --build-arg PULSAR_IMAGE_TAG="$PULSAR_IMAGE_TAG"
+docker build --platform linux/amd64 -t ${GO_SAMPLE} images/samples/go-function-samples --build-arg PULSAR_IMAGE_TAG="$PULSAR_IMAGE_TAG"
 docker tag ${GO_SAMPLE} "${DOCKER_REPO}"/${GO_SAMPLE}:"${PULSAR_IMAGE_TAG}"
 
+echo "build connector sample"
+docker build --platform linux/amd64 -t ${CONNECTOR_ES_SAMPLE} images/samples/pulsar-io-connector/pulsar-io-elasticsearch --build-arg PULSAR_IMAGE_TAG="$PULSAR_IMAGE_TAG" --build-arg PULSAR_IMAGE="$PULSAR_IMAGE" --build-arg RUNNER_IMAGE="$JAVA_RUNNER_IMAGE"
+docker tag ${GO_SAMPLE} "${DOCKER_REPO}"/${CONNECTOR_ES_SAMPLE}:"${PULSAR_IMAGE_TAG}"
```
Comment (imaffe) [inline images/samples/build.sh:48]: Guess it should be `docker tag ${CONNECTOR_ES_SAMPLE} ` here ?

**Review by imaffe — COMMENTED** [review]
Just curious, did these docker images got pushed to remote container registry (seem the es sample connector is not pushed to kind as well)?  How will these containers be used ?

File: images/samples/build.sh (hunk 1 above)
Comment (freeznet) [reply images/samples/build.sh]: nice catch

**Comment by freeznet** [comment]
> Just curious, did these docker images got pushed to remote container registry (seem the es sample connector is not pushed to kind as well)? How will these containers be used ?

this is just act as a sample Dockerfile to show user how to build a connector image.

**Review by imaffe — APPROVED** [review]

**Pull request merged**